pkg/client/client.go

```diff
@@ -734,9 +735,16 @@ func (c *Client) CreateOrUpdateDeployment(dep *appsv1.Deployment) error {
 		uErr, ok := err.(*apierrors.StatusError)
 		if ok && uErr.ErrStatus.Code == 422 && uErr.ErrStatus.Reason == metav1.StatusReasonInvalid {
 			// try to delete Deployment
-			err = c.DeleteDeployment(existing)
-			if err != nil {
-				return errors.Wrap(err, "deleting Deployment object failed")
+			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I'd expect the retry to be on `CreateDeployment()` since it might fail if the existing deployment resource hasn't been fully deleted yet.